generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -832,8 +831,9 @@ public Expr apply(Expr expr, BasicType type, boolean constContext, int depth,
       for (int i = 0; i < numIndices; i++) {
         Expr argument = new ArrayIndexExpr(expr.clone(), new IntConstantExpr(String.valueOf(i)));
         if (i == indexToFurtherTransform) {
-          argument = applyIdentityFunction(argument, type.getElementType(), constContext, depth,
-              fuzzer);
+          argument = applyIdentityFunction(argument,
+              (type.isVector() ? type.getElementType() : type.getColumnType()),
```

Comment:
^ I would remove the parentheses around the ternary, but this is really a matter of taste.